File: PlotFunctions/plotNQz.py
import numpy as np
import matplotlib.pyplot as plt

def plot_nqz(ax, alpha, nqz, mm=None, std=None, z=None, onstd=True):
    """Replicates MATLAB's plotNQz function in Python.

    Parameters
    ----------
    ax : matplotlib.axes.Axes
        Axis on which to draw.
    alpha : array_like
        Sampling parameter values.
    nqz : ndarray
        Matrix with columns representing each quantity to plot.
    mm : ndarray, optional
        Unused placeholder to mirror MATLAB signature.
    std : ndarray, optional
        Standard deviation for each curve. Should match nqz's shape.
    z : sequence, optional
        Unused placeholder to mirror MATLAB signature.
    onstd : bool, default True
        Whether to display the standard deviation bands.
    """
    col = ['r', 'b', 'm', 'g']
    lin = ['-', '--', '-.', ':']
    nqz = np.asarray(nqz)
    n = nqz.shape[1]
    if std is None:
        std = np.zeros_like(nqz)
    else:
        std = np.asarray(std)
    ymin = np.min(nqz)
    ymax = np.max(nqz)
    for i in range(n):
        if onstd:
            ax.fill_between(alpha, nqz[:, i] - std[:, i], nqz[:, i] + std[:, i],
                             color=col[i], edgecolor='white', alpha=0.2)
        ax.plot(alpha, nqz[:, i], color=col[i], linestyle=lin[i], linewidth=2)
    ax.set_ylim([ymin, ymax])
    labels = ['Type-I ERM-RER', 'Type-II ERM-RER', 'Shannon-Jensen', 'Hellinger'][:n]
    ax.legend(labels, loc='best', fontsize=12)
    ax.grid(True)
    ax.set_xlabel(r'$Q(\{L_{\mathbf{z}}(\boldsymbol{\theta})<0.5\})$', fontsize=12)
    return ax.figure.get_size_inches()

File: PlotFunctions/plotNQz.m
function [pos] = plotNQz(h,alpha,NQz,Mm,Std,z,varargin)
     if nargin > 6
        if isa(varargin{1},'logical')
            Onstd = varargin{1};
        else
            Onstd = true;
        end
    else
        Onstd = true;
    end
    col = {'r','b','m','g'};
    lin = {'-','--','-.',':'};
    n = length(NQz(1,:));
    set(h,'Units','Inches');
    pos = get(h,'Position');
    h.Position = [pos(1) pos(2) pos(3)+1 pos(4)];
    ymin = min(NQz,[],'all');
    ymax = max(NQz,[],'all');
    hold on;
    if (Onstd == true)
    for i=1:n
        patch([alpha fliplr(alpha)], [(NQz(:,i)-Std(i))' fliplr((NQz(:,i)+Std(i))')], ...
           col{i}, ...
           'EdgeColor','white');
        alpha(0.2)
    end
    A = { '','','','',...
                     'Type-I ERM-RER',...
                     'Type-II ERM-RER',...
                     'Shannon-Jensen',...
                     'Hellinger'};
    else
    A = {'Type-I ERM-RER',...
         'Type-II ERM-RER',...
         'Shannon-Jensen',...
         'Hellinger'};
    end
    %set(gca, 'XScale', 'log', 'YScale','log')
    for i=1:n
        % semilogx(lamb,R(:,i),col{i},'LineStyle',lin{i},'LineWidth',2);
        plot(alpha,NQz(:,i),col{i},'LineStyle',lin{i},'LineWidth',2);
        ylim([ymin,ymax]);
        if i ==1
            hold on;
        end
    end
    lgd = legend(A,'Interpreter','latex','Location','Best','FontSize',12);
    title(lgd,['$N_{Q,\mbox{\boldmath $z$}}(\lambda)$']);
    % legend({ '','','','',...
    %         ['$R_{\mbox{\boldmath $z$}_',z{2},'}\!\left(P^{(Q,\lambda)}_{\mbox{\boldmath $\Theta$}|\mbox{\boldmath $Z$}=\mbox{\boldmath $z$}_',z{1},'}\right)$'],...
    %         ['$R_{\mbox{\boldmath $z$}_',z{2},'}\!\left(\bar{P}^{(Q,\lambda)}_{\mbox{\boldmath $\Theta$}|\mbox{\boldmath $Z$}=\mbox{\boldmath $z$}_',z{1},'}\right)$']
    %         },'Interpreter','latex','Location','northwest','FontSize',14);
    hold off;
    grid on;
    box on
    xlabel(['$Q(\{L_{\mbox{\boldmath $z$}}(\mbox{\boldmath $\theta$})<0.5\})$'],'Interpreter','latex')

end